Gems/Terrain/Code/Source/TerrainRenderer/TerrainClipmapManager.cpp

```diff
@@ -570,4 +592,101 @@ namespace Terrain
     {
         return m_detailTotalDispatchThreadX != 0 && m_detailTotalDispatchThreadY != 0;
     }
+
+    // AzFramework::Terrain::TerrainDataNotificationBus overrides...
+    void TerrainClipmapManager::OnTerrainDataChanged(
+        [[maybe_unused]] const AZ::Aabb& dirtyRegion, [[maybe_unused]] TerrainDataChangedMask dataChangedMask)
+    {
+        ResetFullRefreshClipmaps();
+    }
```

Comment:
Suggestion: This is probably doing more refreshes than it needs to right now. It should be able to skip full refreshes in the following conditions:
1. dataChangedMask == HEIGHT (i.e. only heights change). 
2. The dirtyRegion is valid and doesn't overlap the clipmap bounding boxes in XY.

Ideally, this would eventually only do partial refreshes where the dirty region overlaps the clipmap, but at least this will get it functionally correct for now.

Reply:
That's correct. There's another optimization that we can have different flags for macro and detail, so when editor changes macro, detail won't update.
But I want to land the correctness before pushing forward. Anyway, updating the whole clipmap in the editor is not that impactful and won't affect game mode or flying the camera.